packages/venia-ui/lib/queries/productSearch.graphql

```diff
@@ -2,21 +2,21 @@ query productSearch(
     $currentPage: Int = 1
     $inputText: String!
     $pageSize: Int = 6
-    $categoryId: String
+    $filters: ProductAttributeFilterInput!
```

Comment:
This is interesting - there is no "Object" type, and locally we have no knowledge of `ProductAttributeFilterInput`. We know this query requires this type though, because of the schema. I just thought it was weird.

Reply:
@jimbo it's the first time I've personally had to write a query on the client that uses a custom type instead of one of the natives such as `Int` or `String`. I just felt weird writing it - like if the server changed the type name the client side would break.

I had to do some custom type matching as well for the individual filter input types and their expected props, which you'll see in the helper.js file.

I'd prefer to use an `Object` type is all I'm saying, but I don't think that exists.